### PR TITLE
Add metarange ID to gc commits

### PR DESCRIPTION
**Pull request by johnnyaug**

### CODE REVIEW

File: clients/spark/core/src/test/scala/io/treeverse/clients/GarbageCollectorSpec.scala [hunk 1]
```diff
@@ -41,6 +43,11 @@ class ARangeGetter(
     }
   }
 
+  def getRangeIDs(metaRangeURL: String): Iterator[String] = {
```
Comment (nopcoder) [inline clients/spark/core/src/test/scala/io/treeverse/clients/GarbageCollectorSpec.scala:46]: Suggestion: getRangeIDsByURL - not a fan of overloading

**Review by nopcoder — CHANGES_REQUESTED** [review]
Can we always add the metarange ID? if we can skip the API change and include always this information it will simplify the code.

File: clients/spark/core/src/test/scala/io/treeverse/clients/GarbageCollectorSpec.scala (hunk 1 above)
Comment (johnnyaug) [reply clients/spark/core/src/test/scala/io/treeverse/clients/GarbageCollectorSpec.scala:46]: I understand the argument against overloading, but I think in this case it does make sense: the method signature explains what it does.

getRangeIDs [by] (**metaRangeURL** : String)
getRangeIDs [by] (**commitID**: String, **repo**: String)

**Review by nopcoder — COMMENTED** [review]
LGTM!

**Review by nopcoder — APPROVED** [review]

**Pull request merged**